homeassistant/components/kdeconnect/__init__.py

```diff
@@ -0,0 +1,107 @@
+from pykdeconnect.devices import KdeConnectDevice
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE_ID, Platform
+from homeassistant.core import HomeAssistant, State
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DATA_KEY_CLIENT, DATA_KEY_DEVICES, DATA_KEY_STORAGE, DOMAIN
+from .helpers import ensure_running
+from .storage import HomeAssistantStorage
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORMS: list[Platform] = [
+    # Platform.BUTTON,
+    Platform.BINARY_SENSOR,
+    Platform.SENSOR,
+    # Platform.DEVICE_TRACKER,
+    # Platform.MEDIA_PLAYER,
+]
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the KDE Connect integration."""
+    await ensure_running(hass)
+
+    if DOMAIN in config:
+        _LOGGER.warning("%s can't be set up using the configuration file", DOMAIN)
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up KDE Connect from a config entry."""
+    client = cast(KdeConnectClient, hass.data[DOMAIN][DATA_KEY_CLIENT])
+    storage = cast(HomeAssistantStorage, hass.data[DOMAIN][DATA_KEY_STORAGE])
+
+    storage.add_device(entry)
+
+    device_id = entry.data[CONF_DEVICE_ID]
+    device = client.get_device(device_id)
+    assert device is not None
+
+    hass.data[DOMAIN][DATA_KEY_DEVICES][entry.entry_id] = device
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        del hass.data[DOMAIN][DATA_KEY_DEVICES][entry.entry_id]
+
+    return unload_ok
+
+
+async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Remove a config entry."""
+    client = cast(KdeConnectClient, hass.data[DOMAIN][DATA_KEY_CLIENT])
+    storage = cast(HomeAssistantStorage, hass.data[DOMAIN][DATA_KEY_STORAGE])
+    device_id = entry.data[CONF_DEVICE_ID]
+    storage.remove_device_by_id(device_id)
+
+    device = client.get_device(device_id)
+    if device is not None and device.is_connected:
+        device.unpair()
+
+
+class KdeConnectEntity(RestoreEntity, ABC):
+    """A base class for all KDE Connect Entities."""
+
+    def __init__(self, device: KdeConnectDevice) -> None:
+        """Initialize the KDE Connect Entity."""
+        self.device = device
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the current device's device info."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.device.device_id)},
+            name=self.device.device_name,
+            manufacturer="KdeConnect",
+        )
```

Comment:
If these values are not expected to change, you can put this in the constructor.

Reply:
Device id can never change. Device name could change, but isn't really expected to change. I've moved it to the constructor for now and will add a callback called when the name is changed in the future.